driver/src/contracts/stablex_contract.rs

```diff
@@ -68,6 +69,7 @@ pub trait StableXContract {
         orders: Vec<Order>,
         solution: Solution,
         claimed_objective_value: U256,
+        get_gas_price: &(dyn GetGasPrice + 'static),
```

Comment:
I had to add `'static` here when compiling in test mode. This seems to be a limitation of the current mockall that might get fixed in the future

Reply:
Can't we make getGasPrice a dependency of the concrete implementation of the contract trait (pass it in new and store it on the struct) rather than passing it on the signature?

To me the way gas price estimation is done seems like an implementation detail of the contract and not something that needs to be part of the interface.

Reply:
We `impl` on `BatchExchange` that comes from the generated `batch_exchange.rs` so there is  no struct to store things in. But we can make our own struct that contains `BatchExchange` and anything else we need to store.